WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java

```diff
@@ -205,6 +205,8 @@ public static void browsePostOrPage(Context context, SiteModel site, PostModel p
 
         // always add the preview parameter to avoid bumping stats when viewing posts
         String url = UrlUtils.appendUrlParameter(post.getLink(), "preview", "true");
+        // Custom domains are not properly authenticated due to a server side(?) issue, so this gets around that
+        url = url.replace(site.getUrl(), site.getUnmappedUrl());
```

Comment:
The unmapped URL can be `null` here (for example, for self hosted sites), and will cause a crash. We should probably also skip this if `getUnmappedUrl()` returns the empty string.